cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/UserTaskManager.java

```diff
@@ -525,6 +519,14 @@ public UserTaskInfo setState(TaskState nextState) {
       _state = nextState;
       return this;
     }
+
+    public boolean isUserTaskDone() {
+      return _futures == null || _futures.isEmpty() || _futures.get(_futures.size() - 1).isDone();
+    }
+
+    public boolean isUserTaskDoneExceptionally() {
+      return _futures != null && !_futures.isEmpty() && _futures.get(_futures.size() - 1).isCompletedExceptionally();
```

Comment:
[Applies to this method and `isUserTaskDone(..)`]: Is it ever expected to have a user task with `null` or empty `_futures`? If this is an illegal state, should we have a sanity check upon creating `UserTaskInfo` with this check -- i.e. `illegalArgumentException`? Also (1) do we need to make these methods `public` -- if so, (2) do we need some synchronization to avoid missing cases where the task makes a transition from active to completed/completed_with_error?